/*
This file holds the Z-Control Feedback messages.
*/

edition = "2023";

import "google/protobuf/timestamp.proto";


message ZCtrlParameters {
  bool feedbackOn = 1;  // Should only be set by MicroscopeTranslator
  float proportionalGain = 2;
  float integralGain = 3;

  // UUID attributes fo associated scan.
  google.protobuf.Timestamp timestamp = 4; // Should match scan timestamp
}


// Linked to Feedback Analysis
message FeedbackAnalysis {
  float proportionOverThreshold = 1;  // In range [0.0, 1.0]
  float proportionUnderThreshold = 2;  // In range [0.0, 1.0]
}
